mintlify/snippets/kyc-onboarding.mdx

```diff
@@ -0,0 +1,632 @@
+
+### Determining Your Platform Status
+
+You can check your platform's regulatory status via the platform configuration:
+
+```http
+GET /config
+```
+
+<ResponseField name="isRegulatedFinancialInstitution" type="boolean">
+Indicates whether your platform is a regulated financial institution. This affects available KYC flows and requirements.
+</ResponseField>
+
+<AccordionGroup>
+<Accordion title="Regulated Platform Example">
+```json
+{
+  "id": "PlatformConfig:019542f5-b3e7-1d02-0000-000000000003",
+  "isRegulatedFinancialInstitution": true,
+  "supportedCurrencies": [
+    {
+      "currencyCode": "USD",
+      "umaProviderRequiredCustomerFields": [
+        "FULL_NAME",
+        "BIRTH_DATE"
+      ]
+    }
+  ]
+}
+```
+</Accordion>
+
+<Accordion title="Unregulated Platform Example">
+```json
+{
+  "id": "PlatformConfig:019542f5-b3e7-1d02-0000-000000000003", 
+  "isRegulatedFinancialInstitution": false,
+  "supportedCurrencies": [
+    {
+      "currencyCode": "USD",
+      "umaProviderRequiredCustomerFields": [
+        "FULL_NAME",
+        "BIRTH_DATE",
+        "NATIONALITY",
+        "ADDRESS",
+        "IDENTITY_DOCUMENT"
+      ]
+    }
+  ]
+}
+```
+</Accordion>
+</AccordionGroup>
+
+## Choosing Your Onboarding Flow
+
+Use this decision tree to determine the best onboarding approach for your platform:
+
+<AccordionGroup>
+<Accordion title="Are you a regulated financial institution?">
+**Yes (isRegulatedFinancialInstitution: true)**
+
+- ✅ **Use Direct API Onboarding** for most customers
+- ✅ Handle KYC/KYB verification internally through your compliance systems
+- ✅ Provide only essential fields required by UMA providers
+- ✅ Faster customer onboarding process
+
+**No (isRegulatedFinancialInstitution: false)**
+
+- ✅ **Use Hosted KYC Link Flow** for new customers
+- ✅ Let UMA providers handle identity verification
+- ✅ Provide comprehensive customer documentation if using direct API
+</Accordion>
+
+<Accordion title="Additional Considerations">
+**When to use Hosted KYC even as a regulated platform:**
+- **Business customers** (always required)
+- Enhanced verification requirements for specific use cases
+- International customers requiring local compliance
+- High-value transactions requiring additional verification
+- Regulatory requirements for specific customer types
+</Accordion>
+</AccordionGroup>
+
+## Hosted KYC Link Flow
+
+The hosted KYC flow provides a secure, hosted interface where customers can complete their identity verification and onboarding process.
+
+<Note>
+**Primary use case**: Unregulated platforms should use this flow for complete identity verification. Regulated platforms may use this flow for enhanced verification or when required by specific use cases. **Business customers must always use this flow regardless of platform regulatory status.**
+</Note>
+
+### Generate KYC Link
+
+<RequestExample>
+```http GET /customers/kyc-link
+GET /customers/kyc-link?redirect_uri=https://yourapp.com/onboarding-complete
```

Comment:
Note for future reference (not needed now): I suspect that users may want to pre-fill some KYC data if possible. For example, if they already have the user's name, perhaps we can pre-fill that, etc.